src/com/dynatrace/loadrunner/Main.java

```diff
@@ -1,59 +1,202 @@
+					System.out.println("Make sure you'd provided correct path to script file \n");
+					return;
+				}
+			} else if (key.equals("-c")) {
+				cEngine = true;
+			} else if (key.equals("-js")) {
+				cEngine = false;
+			} else if (key.equals("body")) {
+				String bodies[] = pair.getValue().split("&");
+				for (String str : bodies) {
+					body.add(new ScriptFile(new File(str)));
+				}
+			} else if (key.equals("header")) {
+				String header[] = pair.getValue().split("&");
+				for (String str : header) {
+					headers.add(new ScriptFile(new File(str)));
+				}
+			} else {
+				showUsages();
+				System.out.println("\nUnknown parameter: " + key);
+				return;
+			}
+		}
+		if (scriptName.isEmpty() && parser.containsKey("path")) {
+			getScriptNameFromPath(parser.arguments.get("path"));
+		}
+		checkEngine();
+		boolean validated = validateParams();
+		if (validated) {
+			LoadRunnerConverter converter = new LoadRunnerConverter();
+			converter.configureConverter(mode, headers, body, scriptName, cEngine);
+			converter.convertFiles();
+			System.out.println("conversion complete");
+		} else {
+			System.out.println("ERROR, conversion failed");
+		}
+	}
+
+	private static void getScriptNameFromPath(String path) {
+		File directory = new File(path);
+		for (File file : directory.listFiles()) {
+			if (getFileExtension(file).equals("usr")) {
+				scriptName = file.getName().replaceFirst("[.][^.]+$", "");
+			}
+		}
+	}
+
+	private static boolean validateParams() {
+		if (headers.isEmpty()) {
+			System.out.println("Headers not found");
+			return false;
+		}
+		if (body.isEmpty()) {
+			System.out.println("Bodies not found");
+			return false;
+		}
+		if (mode == null) {
+			System.out.println("Provide insert/delete parameter");
+			return false;
+		}
+		return true;
+	}
+
+	private static void searchPath(final File folder) {
+		for (final File fileEntry : folder.listFiles()) {
+			if (fileEntry.isDirectory()) {
+				searchPath(fileEntry);
+			}
+			if (fileEntry.getName().equals("globals.js") || fileEntry.getName().equals("globals.h")) {
+				headers.add(new ScriptFile(fileEntry));
+			} else if ((!fileEntry.getName().equals("globals.js") && getFileExtension(fileEntry).equals("js"))
+					|| getFileExtension(fileEntry).equals("c")) {
+				body.add(new ScriptFile(fileEntry));
+			}
+		}
+	}
+
+	private static String getFileExtension(File file) {
+		String fileName = file.getName();
+		if (fileName.lastIndexOf(".") != -1 && fileName.lastIndexOf(".") != 0)
+			return fileName.substring(fileName.lastIndexOf(".") + 1);
+		else
+			return "";
+	}
+
+	private static void getMode(String arg) {
+		if (arg.equals("insert")) {
+			mode = true;
+		}
+		if (arg.equals("delete")) {
+			mode = false;
+		}
+		System.out.println("Setting mode to: " + arg);
+	}
+
+	private static void getScriptName(String name) {
+		scriptName = name;
+		System.out.println("Setting script name to: " + scriptName);
+	}
+
+	private static void checkEngine() {
```

Comment:
I have a very bad associations with the clause "check engine" ;)
call it `validate(InputFileType ift)`